marshal/consumer.go

```diff
@@ -375,6 +394,35 @@ func (c *Consumer) claimTopics() {
 		}
 	}
 
+	// let's compare the new topic claims vs old ones. Upon change, we should trigger an update
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	changed := false
+	// check for missing topic claims
+	for topic := range c.claimedTopics {
+		if !latestClaims[topic] {
+			changed = true
+			break
+		}
+	}
+	// check for new topic claims
+	if !changed && len(latestClaims) != len(c.claimedTopics) {
+		changed = true
+	}
+
+	if changed {
+		c.lock.RUnlock()
```

Comment:
I'm not sure I follow. The call stack is not the only debugging tool we have and I don't think it is worth making the code really hard to read and reason about just to preserve that. Presumably the thing you want out of the call stack pertains to a parameter to one of its parent functions, so just pass that along to identify the request.

I'm not really talking about performance here. I was suggesting you making this anonymous function explicit as a helper to `updateTopicClaims`. I wasn't suggesting that it doesn't need the write lock.